src/me/mrCookieSlime/Slimefun/Setup/SlimefunSetup.java

```diff
@@ -1837,17 +1837,19 @@ public boolean onRightClick(ItemUseEvent e, Player p, ItemStack item) {
 			@Override
 			public boolean onBlockBreak(BlockBreakEvent e, ItemStack item, int fortune, List<ItemStack> drops) {
 				if (SlimefunManager.isItemSimiliar(item, SlimefunItems.PICKAXE_OF_CONTAINMENT, true)) {
-					if (e.getBlock().getType() != Material.SPAWNER) return true;
-					BlockStorage.clearBlockInfo(e.getBlock());
+					Block b = e.getBlock(); // Refactored it into this so we don't need to call e.getBlock() all the time.
+					if (b.getType() != Material.SPAWNER || BlockStorage.hasBlockInfo(b)) return true; 
+					// If the spawner's BlockStorage has BlockInfo, then it's not a vanilla spawner and shouldn't give a broken spawner.
+					BlockStorage.clearBlockInfo(b);
```

Comment:
Ehm, this is quite redundant if you already return when there is BlockInfo.

Reply:
But wouldn't it go through all calls with all sf blocks then?

Reply:
What?

Reply:
I assume the b.getType() != Material.SPAWNER is a check to see if a spawner is broken.
If I remove that and only use the check that I added, BlockStorage.hasBlockInfo(b), then wouldn't it go through with the event call with all non-sf items?

Reply:
!= Material.SPAWNER   makes sure that the block is a spawner.

But this is now what I am commenting on, I made the comment on line 1843.
Clearing the block info is redundant after you checked that there is no block info anyway.

Reply:
oh now I understand what you meant. Lol.

Reply:
Fixed it